packages/node/src/policy-gen/greedy-compartment.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * Provides {@link GreedyCompartment}, which you can just pretend doesn't exist.
+ *
+ * @packageDocumentation
+ */
+
+import { ATTENUATORS_COMPARTMENT } from '../constants.js'
+import { getCanonicalName } from './util.js'
+
+/**
+ * @import {CompartmentOptions, ImportHook, ModuleDescriptor, RecordModuleDescriptor, SourceModuleDescriptor, ModuleSource} from 'ses'
+ * @import {Merge} from 'type-fest'
+ * @import {CompartmentMapDescriptor} from '@endo/compartment-mapper'
+ * @import {LavaMoatPolicyOverrides} from 'lavamoat-core'
+ */
+
+const { entries } = Object
+
+/**
+ * Type guard for a {@link RecordModuleDescriptor} containing a
+ * {@link ModuleSource}
+ *
+ * @param {ModuleDescriptor} moduleDescriptor
+ * @returns {moduleDescriptor is Merge<RecordModuleDescriptor, {record: ModuleSource}>}
+ */
+const isRecordModuleDescriptor = (moduleDescriptor) =>
+  typeof moduleDescriptor === 'object' &&
+  'record' in moduleDescriptor &&
+  moduleDescriptor.record
+
+/**
+ * Type guard for a {@link SourceModuleDescriptor} containing a
+ * {@link ModuleSource}
+ *
+ * @param {ModuleDescriptor} moduleDescriptor
+ * @returns {moduleDescriptor is Merge<SourceModuleDescriptor, {source: ModuleSource}>}
+ */
+const isSourceModuleDescriptor = (moduleDescriptor) =>
+  typeof moduleDescriptor === 'object' &&
+  'source' in moduleDescriptor &&
+  typeof moduleDescriptor.source === 'object' &&
+  moduleDescriptor.source
+
+/**
+ * Returns a constructor for a subclass of `ses`' `Compartment` which injects
+ * package from policy overrides into compartment `imports` if not already
+ * present.
+ *
+ * This causes the new imports to be resolved and expanded as if their usage was
+ * detected. The aim is to cause a generated policy to reflect the state of the
+ * overrides, but not be _limited to_ those overrides (as would likely occur if
+ * we merged them verbatim). This hopefully prevents the end-user from needing
+ * to hand-craft a giant policy override.
+ *
+ * @param {CompartmentMapDescriptor} compartmentMap
+ * @param {LavaMoatPolicyOverrides} [policyOverride]
+ * @returns {typeof Compartment}
+ * @internal
+ */
+export const makeGreedyCompartment = (compartmentMap, policyOverride) => {
+  if (!policyOverride) {
+    return Compartment
+  }
+
+  return class GreedyCompartment extends Compartment {
+    /**
+     * Creates a "greedy" import hook if we can.
+     *
+     * @param {CompartmentOptions} [options]
+     */
+    constructor(options = {}) {
+      const { importHook, name } = options
+
+      if (!importHook || !name || name === ATTENUATORS_COMPARTMENT) {
```

Comment:
- I don't think we need to care about attenuators compartment here (now that we won't even use it)
- Endo should not make us care about the attenuators compartment here. I'm not fully content with how I designed the attenuators compartment's relationship with the rest of the app.

no actions here, unless we wanna remove the `ATTENUATORS_COMPARTMENT` reference.

Reply:
yeah, I don't think `name === ATTENUATORS_COMPARTMENT` will ever be true